plugins/lookup/github_app_access_token.py

```diff
@@ -146,10 +155,14 @@ def run(self, terms, variables=None, **kwargs):
 
         self.set_options(var_options=variables, direct=kwargs)
 
+        if not (self.get_option("key_path") or self.get_option("private_key")):
+            raise AnsibleOptionsError("One of key_path or private_key is required")
```

Comment:
You also need to check whether both are provided (and in that case return an error as well).